components/script/dom/webglrenderingcontext.rs

```diff
@@ -200,6 +201,18 @@ impl WebGLRenderingContext {
     #[allow(unrooted_must_root)]
     pub fn new(window: &Window, canvas: &HTMLCanvasElement, size: Size2D<i32>, attrs: GLContextAttributes)
                -> Option<Root<WebGLRenderingContext>> {
+        if let Some(true) = PREFS.get("webgl.testing.context_creation_error").as_boolean() {
```

Comment:
Let's move this check to `new_inherited`, and `return Err("WebGLRenderingContext creation error forced by pref..".into())` from there, so we don't duplicate all the event handling.